meta-mender-core/classes/mender-artifactcapsule.bbclass

```diff
@@ -0,0 +1,151 @@
+# ------------------------------ CONFIGURATION ---------------------------------
+
+# Extra arguments that should be passed to mender-artifact.
+MENDER_ARTIFACT_CAPSULE_EXTRA_ARGS ?= ""
+
+# The guid used to sign the mender update.
+MENDER_ARTIFACT_SIGNING_KEY ?= ""
+
+do_image_mender_capsule[depends] += "mender-artifact-native:do_populate_sysroot"
+
+MENDER_ARTIFACT_CAPSULE_NAME ??= "${MENDER_ARTIFACT_CAPSULE_NAME_DEFAULT}"
+MENDER_ARTIFACT_CAPSULE_NAME_DEFAULT = "uefi-capsule-${MACHINE}"
+
+MENDER_ARTIFACT_CAPSULE_NAME_DEPENDS ?= ""
+
+MENDER_ARTIFACT_CAPSULE_PROVIDES ?= ""
+
+MENDER_ARTIFACT_CAPSULE_DEPENDS ?= ""
+
+MENDER_ARTIFACT_CAPSULE_CLEAR_PROVIDES ?= ""
+
+CAPSULE_GUIDS ?= ""
+
+CAPSULE_TYPE ?= "uefi-firmware"
+
+
+def get_mender_capsule_artifact_provides(d):
+    # This function creates the provides format required by mender artifact to
+    # generate the capsule artifact.
+    # MENDER_ARTIFACT_CAPSULE_PROVIDES needs to be used in the following way.
+    # MENDER_ARTIFACT_CAPSULE_PROVIDES[e2bb9c06-70e9-4b14-97a3-5a7913176111] = "tfa:3"
+    # MENDER_ARTIFACT_CAPSULE_PROVIDES[e2bb9c06-70e9-4b14-97a3-5a7913176222] = "edk2:3"
+    # This gets converted to the  following provides format:
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176111.name":"tfa",
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176111.version":"3",
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176222.name":"edk2",
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176222.version":"3",
+    guids = d.getVarFlags("MENDER_ARTIFACT_CAPSULE_PROVIDES")
+    capsule_type = d.getVar("CAPSULE_TYPE", expand=True)
+    if guids is None:
+        return
+    provides = ""
+    for guid in guids:
+        splt = guids[guid].split(":")
+        provides += " --provides " + capsule_type + "." + guid + ".version:" + splt[1]
+        provides += " --provides " + capsule_type + "." + guid + ".name:" + splt[0]
+    return provides
+
+def get_mender_capsule_artifact_depends(d):
+    # This function creates the depends format required by mender artifact to
+    # generate the capsule artifact.
+    # MENDER_ARTIFACT_CAPSULE_DEPENDS needs to be used in the following way.
+    #MENDER_ARTIFACT_CAPSULE_DEPENDS[e2bb9c06-70e9-4b14-97a3-5a7913176111] = "2"
+    #MENDER_ARTIFACT_CAPSULE_DEPENDS[e2bb9c06-70e9-4b14-97a3-5a7913176222] = "2"
```

Comment:
I meant that the parsing would also be changed to accommodate it. It's not very important though, I just thought it might be easier to read. I leave it for you to decide.